audit/logger.go

```diff
@@ -77,32 +77,38 @@ type logger struct {
 }
 
 func (l *logger) Action(action string) Logger {
-	return l.addField("action", action)
+	return l.newLoggerWith("action", action)
 }
 
 func (l *logger) Message(ctx datastore.Context, message string) Logger {
-	l.addField("user", ctx.User())
 	l.message = message
-	return l
+	return l.newLoggerWith("user", ctx.User())
```

Comment:
Setting message first may 'break' the notion here that you get a new object. Instead, you can change the return type of `newLoggerWith` to be `*logger`, then you can could do:
```
newLogger := l.newLoggerWith("user", ctx.User())
newLogger.message = message
return newLogger
```